File: src/Fable.Repl.Lib/Feliz.Engine/SvgEngine.fs
namespace Feliz

open System

type SvgEngine<'Node>
    /// <summary>Customizable SVG generator API.</summary>
    ///
    /// <param name="mk">Make a node with tag name and children.</param>
    /// <param name="ofStr">Make a node from a string.</param>
    /// <param name="empty">Make an empty node.</param>
    (mk: string -> 'Node seq -> 'Node, ofStr: string -> 'Node, empty: unit -> 'Node) =

    /// Create a custom element
    ///
    /// You generally shouldn't need to use this, if you notice an element missing please submit an issue.
    member _.custom (key: string, children: seq<'Node>) = mk key children
    /// The empty element, renders nothing on screen
    member _.none : 'Node = empty()

    /// SVG Image element, not to be confused with HTML img alias.
    member _.image (children: seq<'Node>) = mk "image" children
    /// The svg element is a container that defines a new coordinate system and viewport. It is used as the outermost element of SVG documents, but it can also be used to embed an SVG fragment inside an SVG or HTML document.
    member _.svg (children: seq<'Node>) = mk "svg" children
    member _.circle (children: seq<'Node>) = mk "circle" children
    member _.clipPath (children: seq<'Node>) = mk "clipPath" children
    /// The <defs> element is used to store graphical objects that will be used at a later time. Objects created inside a <defs> element are not rendered directly. To display them you have to reference them (with a <use> element for example). https://developer.mozilla.org/en-US/docs/Web/SVG/Element/defs
    member _.defs (children: seq<'Node>) = mk "defs" children
    /// The <desc> element provides an accessible, long-text description of any SVG container element or graphics element.
    member _.desc (value: string) = mk "desc" [ofStr value]
    member _.ellipse (children: seq<'Node>) = mk "ellipse" children
    member _.feBlend (children: seq<'Node>) = mk "feBlend" children
    member _.feColorMatrix (children: seq<'Node>) = mk "feColorMatrix" children
    member _.feComponentTransfer (children: seq<'Node>) = mk "feComponentTransfer" children
    member _.feComposite (children: seq<'Node>) = mk "feComposite" children
    member _.feConvolveMatrix (children: seq<'Node>) = mk "feConvolveMatrix" children
    member _.feDiffuseLighting (children: seq<'Node>) = mk "feDiffuseLighting" children
    member _.feDisplacementMap (children: seq<'Node>) = mk "feDisplacementMap" children
    member _.feDistantLight (children: seq<'Node>) = mk "feDistantLight" children
    member _.feDropShadow (children: seq<'Node>) = mk "feDropShadow" children
    member _.feFlood (children: seq<'Node>) = mk "feFlood" children
    member _.feFuncA (children: seq<'Node>) = mk "feFuncA" children
    member _.feFuncB (children: seq<'Node>) = mk "feFuncB" children
    member _.feFuncG (children: seq<'Node>) = mk "feFuncG" children
    member _.feFuncR (children: seq<'Node>) = mk "feFuncR" children
    member _.feGaussianBlur (children: seq<'Node>) = mk "feGaussianBlur" children
    member _.feImage (children: seq<'Node>) = mk "feImage" children
    member _.feMerge (children: seq<'Node>) = mk "feMerge" children
    member _.feMergeNode (children: seq<'Node>) = mk "feMergeNode" children
    member _.feMorphology (children: seq<'Node>) = mk "feMorphology" children
    member _.feOffset (children: seq<'Node>) = mk "feOffset" children
    member _.fePointLight (children: seq<'Node>) = mk "fePointLight" children
    member _.feSpecularLighting (children: seq<'Node>) = mk "feSpecularLighting" children
    member _.feSpotLight (children: seq<'Node>) = mk "feSpotLight" children
    member _.feTile (children: seq<'Node>) = mk "feTile" children
    member _.feTurbulence (children: seq<'Node>) = mk "feTurbulence" children
    member _.filter (children: seq<'Node>) = mk "filter" children
    member _.foreignObject (children: seq<'Node>) = mk "foreignObject" children
    /// The <g> SVG element is a container used to group other SVG elements.
    ///
    /// Transformations applied to the <g> element are performed on its child elements, and its attributes are inherited by its children. It can also group multiple elements to be referenced later with the <use> element.
    member _.g (children: seq<'Node>) = mk "g" children
    member _.line (children: seq<'Node>) = mk "line" children
    member _.linearGradient (children: seq<'Node>) = mk "linearGradient" children
    /// The <marker> element defines the graphic that is to be used for drawing arrowheads or polymarkers on a given <path>, <line>, <polyline> or <polygon> element.
    member _.marker (children: seq<'Node>) = mk "marker" children
    member _.mask (children: seq<'Node>) = mk "marker" children
    member _.mpath (children: seq<'Node>) = mk "mpath" children
    member _.path (children: seq<'Node>) = mk "path" children
    member _.pattern (children: seq<'Node>) = mk "pattern" children
    member _.polygon (children: seq<'Node>) = mk "polygon" children
    member _.polyline (children: seq<'Node>) = mk "polyline" children
    member _.set (children: seq<'Node>) = mk "set" children
    member _.stop (children: seq<'Node>) = mk "stop" children
    member _.style (value: string) = mk "style" [ofStr value]
    member _.switch (children: seq<'Node>) = mk "switch" children
    member _.symbol (children: seq<'Node>) = mk "symbol" children
    member _.text (content: string) = mk "text" [ofStr content]
    member _.title (content: string) = mk "title" [ofStr content]
    member _.textPath (children: seq<'Node>) = mk "textPath" children
    member _.tspan (children: seq<'Node>) = mk "tspan" children
    member _.use' (children: seq<'Node>) = mk "use" children
    member _.radialGradient (children: seq<'Node>) = mk "radialGradient" children
    member _.rect (children: seq<'Node>) = mk "rect" children
    member _.view (children: seq<'Node>) = mk "view" children

File: js/repl/fable-repl-lib/Feliz.Engine/SvgEngine.js
import { class_type } from "../../fable-library/Reflection.js";
import { singleton } from "../../fable-library/List.js";

export class Feliz_SvgEngine$1 {
    constructor(mk, ofStr, empty) {
        this.mk = mk;
        this.ofStr = ofStr;
        this.empty = empty;
    }
}

export function Feliz_SvgEngine$1$reflection(gen0) {
    return class_type("Feliz.SvgEngine`1", [gen0], Feliz_SvgEngine$1);
}

export function Feliz_SvgEngine$1_$ctor_Z780DA98D(mk, ofStr, empty) {
    return new Feliz_SvgEngine$1(mk, ofStr, empty);
}

export function Feliz_SvgEngine$1__custom_4AE854A1(_, key, children) {
    return _.mk(key, children);
}

export function Feliz_SvgEngine$1__get_none(_) {
    return _.empty();
}

export function Feliz_SvgEngine$1__image_BB573A(_, children) {
    return _.mk("image", children);
}

export function Feliz_SvgEngine$1__svg_BB573A(_, children) {
    return _.mk("svg", children);
}

export function Feliz_SvgEngine$1__circle_BB573A(_, children) {
    return _.mk("circle", children);
}

export function Feliz_SvgEngine$1__clipPath_BB573A(_, children) {
    return _.mk("clipPath", children);
}

export function Feliz_SvgEngine$1__defs_BB573A(_, children) {
    return _.mk("defs", children);
}

export function Feliz_SvgEngine$1__desc_Z721C83C5(_, value) {
    const arg10 = singleton(_.ofStr(value));
    return _.mk("desc", arg10);
}

export function Feliz_SvgEngine$1__ellipse_BB573A(_, children) {
    return _.mk("ellipse", children);
}

export function Feliz_SvgEngine$1__feBlend_BB573A(_, children) {
    return _.mk("feBlend", children);
}

export function Feliz_SvgEngine$1__feColorMatrix_BB573A(_, children) {
    return _.mk("feColorMatrix", children);
}

export function Feliz_SvgEngine$1__feComponentTransfer_BB573A(_, children) {
    return _.mk("feComponentTransfer", children);
}

export function Feliz_SvgEngine$1__feComposite_BB573A(_, children) {
    return _.mk("feComposite", children);
}

export function Feliz_SvgEngine$1__feConvolveMatrix_BB573A(_, children) {
    return _.mk("feConvolveMatrix", children);
}

export function Feliz_SvgEngine$1__feDiffuseLighting_BB573A(_, children) {
    return _.mk("feDiffuseLighting", children);
}

export function Feliz_SvgEngine$1__feDisplacementMap_BB573A(_, children) {
    return _.mk("feDisplacementMap", children);
}

export function Feliz_SvgEngine$1__feDistantLight_BB573A(_, children) {
    return _.mk("feDistantLight", children);
}

export function Feliz_SvgEngine$1__feDropShadow_BB573A(_, children) {
    return _.mk("feDropShadow", children);
}

export function Feliz_SvgEngine$1__feFlood_BB573A(_, children) {
    return _.mk("feFlood", children);
}

export function Feliz_SvgEngine$1__feFuncA_BB573A(_, children) {
    return _.mk("feFuncA", children);
}

export function Feliz_SvgEngine$1__feFuncB_BB573A(_, children) {
    return _.mk("feFuncB", children);
}

export function Feliz_SvgEngine$1__feFuncG_BB573A(_, children) {
    return _.mk("feFuncG", children);
}

export function Feliz_SvgEngine$1__feFuncR_BB573A(_, children) {
    return _.mk("feFuncR", children);
}

export function Feliz_SvgEngine$1__feGaussianBlur_BB573A(_, children) {
    return _.mk("feGaussianBlur", children);
}

export function Feliz_SvgEngine$1__feImage_BB573A(_, children) {
    return _.mk("feImage", children);
}

export function Feliz_SvgEngine$1__feMerge_BB573A(_, children) {
    return _.mk("feMerge", children);
}

export function Feliz_SvgEngine$1__feMergeNode_BB573A(_, children) {
    return _.mk("feMergeNode", children);
}

export function Feliz_SvgEngine$1__feMorphology_BB573A(_, children) {
    return _.mk("feMorphology", children);
}

export function Feliz_SvgEngine$1__feOffset_BB573A(_, children) {
    return _.mk("feOffset", children);
}

export function Feliz_SvgEngine$1__fePointLight_BB573A(_, children) {
    return _.mk("fePointLight", children);
}

export function Feliz_SvgEngine$1__feSpecularLighting_BB573A(_, children) {
    return _.mk("feSpecularLighting", children);
}

export function Feliz_SvgEngine$1__feSpotLight_BB573A(_, children) {
    return _.mk("feSpotLight", children);
}

export function Feliz_SvgEngine$1__feTile_BB573A(_, children) {
    return _.mk("feTile", children);
}

export function Feliz_SvgEngine$1__feTurbulence_BB573A(_, children) {
    return _.mk("feTurbulence", children);
}

export function Feliz_SvgEngine$1__filter_BB573A(_, children) {
    return _.mk("filter", children);
}

export function Feliz_SvgEngine$1__foreignObject_BB573A(_, children) {
    return _.mk("foreignObject", children);
}

export function Feliz_SvgEngine$1__g_BB573A(_, children) {
    return _.mk("g", children);
}

export function Feliz_SvgEngine$1__line_BB573A(_, children) {
    return _.mk("line", children);
}

export function Feliz_SvgEngine$1__linearGradient_BB573A(_, children) {
    return _.mk("linearGradient", children);
}

export function Feliz_SvgEngine$1__marker_BB573A(_, children) {
    return _.mk("marker", children);
}

export function Feliz_SvgEngine$1__mask_BB573A(_, children) {
    return _.mk("marker", children);
}

export function Feliz_SvgEngine$1__mpath_BB573A(_, children) {
    return _.mk("mpath", children);
}

export function Feliz_SvgEngine$1__path_BB573A(_, children) {
    return _.mk("path", children);
}

export function Feliz_SvgEngine$1__pattern_BB573A(_, children) {
    return _.mk("pattern", children);
}

export function Feliz_SvgEngine$1__polygon_BB573A(_, children) {
    return _.mk("polygon", children);
}

export function Feliz_SvgEngine$1__polyline_BB573A(_, children) {
    return _.mk("polyline", children);
}

export function Feliz_SvgEngine$1__set_BB573A(_, children) {
    return _.mk("set", children);
}

export function Feliz_SvgEngine$1__stop_BB573A(_, children) {
    return _.mk("stop", children);
}

export function Feliz_SvgEngine$1__style_Z721C83C5(_, value) {
    const arg10 = singleton(_.ofStr(value));
    return _.mk("style", arg10);
}

export function Feliz_SvgEngine$1__switch_BB573A(_, children) {
    return _.mk("switch", children);
}

export function Feliz_SvgEngine$1__symbol_BB573A(_, children) {
    return _.mk("symbol", children);
}

export function Feliz_SvgEngine$1__text_Z721C83C5(_, content) {
    const arg10 = singleton(_.ofStr(content));
    return _.mk("text", arg10);
}

export function Feliz_SvgEngine$1__title_Z721C83C5(_, content) {
    const arg10 = singleton(_.ofStr(content));
    return _.mk("title", arg10);
}

export function Feliz_SvgEngine$1__textPath_BB573A(_, children) {
    return _.mk("textPath", children);
}

export function Feliz_SvgEngine$1__tspan_BB573A(_, children) {
    return _.mk("tspan", children);
}

export function Feliz_SvgEngine$1__use$0027_BB573A(_, children) {
    return _.mk("use", children);
}

export function Feliz_SvgEngine$1__radialGradient_BB573A(_, children) {
    return _.mk("radialGradient", children);
}

export function Feliz_SvgEngine$1__rect_BB573A(_, children) {
    return _.mk("rect", children);
}

export function Feliz_SvgEngine$1__view_BB573A(_, children) {
    return _.mk("view", children);
}

